### DISCUSSION TITLE
Get test name inside of test.

### DISCUSSION THREAD

**Discussion by jwidauer** [onqtam/doctest#806] [Q&A]
I've been loving doctest so far, but have come across an interesting question.
Is there a way to get the currently running tests name while inside a test?
So something like this:

```cpp
TEST_CASE("case") {
  SUBCASE("subcase") {
    std::cout << doctest::test_name() << std::endl;
  }
}
```

or even:

```cpp
TEST_CASE("case") {
  std::cout << doctest::test_name() << std::endl;
  SUBCASE("subcase1") {
    // do some testing
  }
  SUBCASE("subcase2") {
    // do some more testing
  }
}
```

What I would like to be the output there, is some string that is unique to the currently running test case / subcase.
So in the first example something along the lines of `case.subcase`.
And in the second example something along the lines of `case.subcase1` when running the first subcase and `case.subcase2` when running the second subcase.

Is something like that supported?
If not, would this be a feasible enhancement?

**Comment by onqtam** [comment]
Currently, the only thing you can get is the current test case name like this:
https://github.com/doctest/doctest/issues/345#issuecomment-593625010
Perhaps you could hack your way around getting the subcase name too. This should be implemented at some point, but the project is looking for maintainers: https://github.com/doctest/doctest/issues/554
